### PR TITLE
Add unit tests

### LINKED ISSUE #161: Add unit tests
Add unit tests so we have bare bones testing functionality.

### PR DESCRIPTION
**Pull request by akaktus**
[-] Added unit tests for viewModels.
Fixes #161

### CODE REVIEW

File: O-FISHTests/AnnotatedNoteViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        //given
+        let note = "Simplet Note"
+        let annotatedNote = AnnotatedNote()
+        annotatedNote.note = note
+        
+        //when
+        sut = AnnotatedNoteViewModel(annotatedNote)
+        
+        //then
+        XCTAssertNotNil(sut)
```
Comment (kojiba) [inline O-FISHTests/AnnotatedNoteViewModelTests.swift:79]: compare note from model is equal to note in viewmodel

File: O-FISHTests/BoatViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        sut?.ems.append(EMSViewModel.sample)
+        
+        //when
+        let boat = sut?.save()
+        
+        //then
+        XCTAssertEqual(boat?.name, name)
+        XCTAssertEqual(boat?.homePort, homePort)
+        XCTAssertEqual(boat?.nationality, nationality)
+        XCTAssertEqual(boat?.permitNumber, permitNumber)
```
Comment (kojiba) [inline O-FISHTests/BoatViewModelTests.swift:73]: if we adding EMS we also should test EMS for equality

File: O-FISHTests/CatchViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        let quantityType: [CatchViewModel.QuantityType] = [.count, .weight]
+        
+        //when
+        sut?.fish = fish
+        sut?.number = number
+        sut?.weight = weight
+        sut?.unit = unit
+        sut?.quantityType = quantityType
+        let catchFish = sut?.save()
+
```
Comment (kojiba) [inline O-FISHTests/CatchViewModelTests.swift:94]: too much newlines

File: O-FISHTests/CrewMemberViewModelTests.swift [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        sut = nil
+    }
+    
+    func testIsEmptyTrue() {
+        let isEmpty = sut?.isEmpty ?? false
+        XCTAssertTrue(isEmpty)
+    }
+    
+    func testIsEmptyFalse() {
+
```
Comment (kojiba) [inline O-FISHTests/CrewMemberViewModelTests.swift:29]: extra newline

File: O-FISHTests/DeliveryViewModelTests.swift [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        let delivery: Delivery? = nil
+        sut = DeliveryViewModel(delivery)
+        
+        //when
+        let deliveryObj = sut?.save()
+        
+        //then
+        XCTAssertNotNil(deliveryObj)
+    }
+
```
Comment (kojiba) [inline O-FISHTests/DeliveryViewModelTests.swift:78]: extra newlines

File: O-FISHTests/EMSViewModelTests.swift [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        sut = nil
+    }
+    
+    func testIsEmptyTrue() {
+        let isEmpty = sut?.isEmpty ?? false
+        XCTAssertTrue(isEmpty)
+    }
+    
+    func testIsEmptyFalse() {
+
```
Comment (kojiba) [inline O-FISHTests/EMSViewModelTests.swift:29]: extra newline

File: O-FISHTests/OffenceViewModelTests.swift [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    
+    func testOffenceIsEmptyFalse() {
+        //when
+        let isEmpty = sut?.isEmpty ?? true
+        
+        //then
+        XCTAssertTrue(isEmpty)
+    }
+
```
Comment (kojiba) [inline O-FISHTests/OffenceViewModelTests.swift:48]: extra newlines

File: O-FISHTests/ViolationViewModelTests.swift [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        //when
+        let string = sut?.fullViolationDescription
+        
+        //then
+        XCTAssertNotNil(string)
+    }
+    
+    func testFullViolationDescriptionEmpty() {
+        //given
+
```
Comment (kojiba) [inline O-FISHTests/ViolationViewModelTests.swift:43]: extra newline

File: o-fish-ios/ViewModel/Report/DeliveryViewModel.swift [hunk 9]
```diff
@@ -47,5 +47,5 @@ class DeliveryViewModel: ObservableObject, Identifiable {
         return delivery
     }
 
-    var isEmtpy: Bool { date == nil && location.isEmpty && business.isEmpty }
+    var isEmpty: Bool { date == nil && location.isEmpty && business.isEmpty }
```
Comment (kojiba) [inline o-fish-ios/ViewModel/Report/DeliveryViewModel.swift:50]: what actually changed?

Comment (akaktus) [reply o-fish-ios/ViewModel/Report/DeliveryViewModel.swift:50]: mistype. was isEm"t"py, should "isEmpty"

**Pull request merged**